io-engine/src/lvs/mod.rs

```diff
@@ -192,8 +192,13 @@ impl IPoolProps for Lvs {
             .crypto_base_bdev()
             .map(Bdev::new)
             .unwrap_or_else(|| self.base_bdev());
-
-        vec![disk_bdev.bdev_uri_str().unwrap_or_else(|| "".into())]
+        let bdev_uri = disk_bdev.bdev_uri_str().unwrap_or_else(|| "".into());
+        let bdev_path = if bdev_uri.starts_with("malloc") {
+            bdev_uri.split("&").next().unwrap_or("")
+        } else {
+            bdev_uri.split("?").next().unwrap_or("")
+        };
```

Comment:
Use scheme() getter I suggest.

```suggestion
        let bdev_path = match bdev_uri.scheme()  {
            "malloc" => bdev_uri.split("&").next().unwrap_or(""),
            _ => bdev_uri.split("?").next().unwrap_or("")
        }
```

Reply:
ok, if it's a string, it'll not work.

Reply:
Yes, will have to parse it to Url. Easier this way.